### PR TITLE
Bump guice to v6.0.0 and guava to v32.1.1

### PR DESCRIPTION
**Pull request by ThePumpingLemma**
Additionally, use BOMs for both guice and guava

### CODE REVIEW

File: misk-aws-dynamodb-testing/build.gradle.kts [hunk 1]
```diff
@@ -10,11 +10,13 @@ plugins {
 
 dependencies {
   api(Dependencies.awsDynamodb)
+  api(Dependencies.guava)
   api(Dependencies.guice)
   api(Dependencies.javaxInject)
   api(Dependencies.tempestTestingInternal)
   api(project(":misk-aws-dynamodb"))
   api(project(":misk-inject"))
+  implementation(Dependencies.errorproneAnnotations)
```
Comment (ThePumpingLemma) [inline misk-aws-dynamodb-testing/build.gradle.kts:19]: I think we can actually omit these if we want, see: https://github.com/google/guice/pull/1739

**Review by chris-ryan-square — APPROVED** [review]

File: misk-aws-dynamodb-testing/build.gradle.kts (hunk 1 above)
Comment (adrw) [reply misk-aws-dynamodb-testing/build.gradle.kts:19]: If we don't need it, let's remove it. 

Could be worth running the dependency analyzer plugin again, last time it caught stuff like this.

**Review by adrw — APPROVED** [review]
LG

File: misk-aws-dynamodb-testing/build.gradle.kts (hunk 1 above)
Comment (ThePumpingLemma) [reply misk-aws-dynamodb-testing/build.gradle.kts]: I pushed a change to omit it, let's see if the tests pass. 🤞

**Review by keeferrourke — APPROVED** [review]

**Review by frojasg — APPROVED** [review]
LGTM. Thanks for doing this update

**Pull request merged**